Rewrite: modular approach

ghostbar opened the pull request:
Fixes #3

ghostbar committed:
feat(gh): move github-related code to gh

ghostbar committed:
feat(tg): move telegram-related code to tg

ghostbar committed:
feat(handler): move handler code to handler pkg

ghostbar committed:
refac(now.json): point to right entry

Now the entrypoint is `handler/handler.go`.

sadasant commented on gh/content.go:
```diff
@@ -0,0 +1,39 @@
+// NotAllowed returns an error if the received Action is not valid.
+func (c Content) NotAllowed() error {
+	switch c.Action {
+	case "labeled",
```
Awesome!

ghostbar replied:
:-D

sadasant commented on gh/comment.go:
```diff
@@ -0,0 +1,17 @@
+package gh
```
Can you make this more explicit? as in `github`? Same with `telegram`. Please 🙏❤️

ghostbar replied:
The github library already calls itself github. That would create some issues. Better avoid it and leave it as `gh` and `tg` :-)

ghostbar committed:
docs(CHANGELOG): initial commit

sadasant commented:
This is the problem that CircleCI is having: https://github.com/golang/go/issues/27056
I'm also having this myself. On it...

sadasant committed:
module files and CI bash scripts

- deleted the handler_test for now
- missing modules
- CI bash scripts

Merged and squashed from: https://github.com/berserktech/telebot/pull/8

sadasant committed:
mod-update.bash, to update local references in our go.mods

sadasant committed:
one last run of mod-update with HEAD to fix CircleCI

sadasant committed:
cleaned up a TODO comment in comment.go

sadasant committed:
feature: synchronized actions not allowed

sadasant committed:
updated local modules to the latest hash

sadasant commented:
IMPORTANT: This doesn't work on Zeit.co. They use an older version:
https://github.com/zeit/now-builders/blob/master/packages/now-go/go-helpers.js#L93
and they disable GO111MODULE by default due to having `GO111MODULE=auto`.

I can manually set `GO111MODULE` to `on`, but 1.11 doesn't know how to mix go.mod with dependencies in GOPATH, see: https://golang.org/doc/go1.12#modules

> ... the go command now supports module-aware operations outside of a module directory, provided that those operations do not need to resolve import paths relative to the current directory or explicitly edit the go.mod file. ...

I did the silliest of pull requests to [zeit/now-builders](https://github.com/zeit/now-builders) just to bring awareness to the issue: https://github.com/zeit/now-builders/pull/247

sadasant committed:
GO111MODULE=on on now.json

ghostbar committed:
refac(mod): back to single-module for zeit

ghostbar commented:
Going back to single-module repo for Zeit :-)

ghostbar committed:
refac(circleci): simplify commands

sadasant committed:
zeit is back

Zeit.co needs a main package with the Handler exposed, so I moved
handler/handler.go to index.go :) now we know.

ghostbar committed:
fix(index): add empty main function

sadasant reviewed (approved):
🎉

sadasant committed:
Updated the README and the CHANGELOG

ghostbar committed:
fix(index): remove empty main func

ghostbar committed:
Merge branch 'feat/rewrite' of github.com:berserktech/telebot into feat/rewrite

sadasant committed:
fixed the synchronize filter

betacar reviewed (approved):
💯
